morepath/dispatch.py

```diff
@@ -0,0 +1,73 @@
+import reg
+from functools import wraps, partial
+from .reify import reify
+
+
+class delegate(object):
+
+    def __init__(self, *predicates):
+        self.make_generic = reg.dispatch(*predicates)
+        self.external_predicates = False
+
+    def __call__(self, func):
+
+        def install(reg):
+            setattr(reg, func.__name__, self.make_generic(func))
+
+        @wraps(func)
+        @reify
+        def delegator(app):
```

Comment:
This layer does bother me; it hides the underlying Reg dispatch API under another thing. On the dispatchmethods-decoupled branch this is not needed, as you have access to the true dispatch object directly.